// Copyright 2017 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package ray.rpc;

enum AgentRpcStatus {
  // OK.
  AGENT_RPC_STATUS_OK = 0;
  // Failed.
  AGENT_RPC_STATUS_FAILED = 1;
}

message RegisterAgentRequest {
  int32 agent_pid = 1;
  int32 agent_port = 2;
  string agent_ip_address = 3;
}

message RegisterAgentReply {
  AgentRpcStatus status = 1;
}

service AgentManagerService {
  rpc RegisterAgent(RegisterAgentRequest) returns (RegisterAgentReply);
}
